File: 2k-171/src/main/java/CalculatorInputParser.java
import java.util.ArrayList;

public class CalculatorInputParser extends AbstractParser {
    @Override
    public ArrayList<String> parse(ArrayList<String> data) {
        validate(data);
        return data;
    }

    @Override
    protected void validate(ArrayList<String> data) {
        if (data.size() != 3) {
            throw new IllegalArgumentException("Incorrect input: 3 arguments must be provided");
        }
        try {
            Double.parseDouble(data.get(0));
        }
        catch (NumberFormatException e) {
            throw new IllegalArgumentException("Incorrect input: the first argument must be a number");
        }
        try {
            Double.parseDouble(data.get(2));
        }
        catch (NumberFormatException e) {
            throw new IllegalArgumentException("Incorrect input: the third argument must be a number");
        }
    }
}

File: 2k-227/src/main/kotlin/CalculatorInputParser.kt

import java.lang.NumberFormatException
import java.util.ArrayList
import kotlin.IllegalArgumentException

class CalculatorInputParser : AbstractParser() {
    override fun parse(data: ArrayList<String>): ArrayList<String> {
        validate(data)
        return data
    }

    override fun validate(data: ArrayList<String>) {
        if (data.size != 3) { throw IllegalArgumentException("Incorrect input: 3 arguments must be provided") }
        try {
            data[0].toDouble()
        } catch (e: NumberFormatException) {
            throw IllegalArgumentException("Incorrect input: the first argument must be a number")
        }
        try {
            data[2].toDouble()
        } catch (e: NumberFormatException) {
            throw IllegalArgumentException("Incorrect input: the third argument must be a number")
        }
    }
}